### PR TITLE
Push some nontested nonsense

### PR DESCRIPTION
**Pull request by romeq**
Reassigning `caps` so many times annoyed me so I decided to make this PR.
I also moved `.await?`s to same row where their functions were.

I have no damn idea if whole parser compiles as I didn't even try to compile (or test) it.

### CODE REVIEW

File: src/util/parse_lykeion.rs [hunk 1]
```diff
@@ -13,24 +13,26 @@ pub async fn parse_lykeion() -> Result<Vec<(NaiveDate, String)>, anyhow::Error>
     for i in 1..=7 {
         let resp = r_client
             .get(format!("https://aromimenu.cgisaas.fi/VaasaAromieMenus/fi-FI/Default/_/CampusLykeion/Rss.aspx?Id=0c9160c7-bedb-4b60-9ee1-188bf43a02b3&DateMode={}", i))
-            .send()
-            .await?
-            .bytes()
-            .await?;
+            .send().await?
+            .bytes().await?;
+
```
Comment (lajp) [inline src/util/parse_lykeion.rs:18]: why are you committing blank lines

File: src/util/parse_lykeion.rs [hunk 2]
```diff
@@ -13,24 +13,26 @@ pub async fn parse_lykeion() -> Result<Vec<(NaiveDate, String)>, anyhow::Error>
...
             .get(format!("https://aromimenu.cgisaas.fi/VaasaAromieMenus/fi-FI/Default/_/CampusLykeion/Rss.aspx?Id=0c9160c7-bedb-4b60-9ee1-188bf43a02b3&DateMode={}", i))
-            .send()
-            .await?
-            .bytes()
-            .await?;
+            .send().await?
+            .bytes().await?;
+        
         let channel = Channel::read_from(&resp[..])?;
+
```
Comment (lajp) [inline src/util/parse_lykeion.rs:20]: wtf, this too

File: src/util/parse_lykeion.rs [hunk 3]
```diff
@@ -13,24 +13,26 @@ pub async fn parse_lykeion() -> Result<Vec<(NaiveDate, String)>, anyhow::Error>
...
             let foodstr = item.description.unwrap();
-            let caps = FOOD_REGEX.captures(&foodstr).unwrap();
-            let caps = caps.get(1).unwrap().as_str();
-            let caps = ALLERGEENI_REGEX.replace_all(caps, ",");
-            let caps = caps.replace("M,G,V", "");
-            let caps = caps.replace("., ", "");
-            let caps = caps.replace(" ,", ",");
-            let caps = caps.replace(",,", ",");
-            let caps = caps.replace("  ", " ");
+            let caps =
```
Comment (lajp) [inline src/util/parse_lykeion.rs]: why is this twice

File: src/util/parse_lykeion.rs [hunk 4]
```diff
@@ -13,24 +13,26 @@ pub async fn parse_lykeion() -> Result<Vec<(NaiveDate, String)>, anyhow::Error>
...
+            let caps = ALLERGEENI_REGEX.replace_all(FOOD_REGEX
+                .captures(&foodstr).unwrap()
+                .get(1).unwrap()
+                .as_str(), ",")
+                .replace("M,G,V", "")
+                .replace("., ", "")
+                .replace(" ,", ",")
+                .replace(",,", ",")
+                .replace("  ", " ");
+
```
Comment (lajp) [inline src/util/parse_lykeion.rs:35]: this is also redundant

**Review by lajp — CHANGES_REQUESTED** [review]
Won't compiler

File: src/util/parse_lykeion.rs (hunk 4 above)
Comment (romeq) [reply src/util/parse_lykeion.rs:35]: mikä vittu on redundant

File: src/util/parse_lykeion.rs (hunk 1 above)
Comment (romeq) [reply src/util/parse_lykeion.rs:18]: they make it nicer

File: src/util/parse_lykeion.rs (hunk 2 above)
Comment (romeq) [reply src/util/parse_lykeion.rs:20]: yes

File: src/util/parse_lykeion.rs (hunk 3 above)
Comment (romeq) [reply src/util/parse_lykeion.rs]: i forgot it

Comment (romeq) [reply src/util/parse_lykeion.rs]: don't judge

File: src/util/parse_lykeion.rs [hunk 5]
```diff
@@ -1,36 +1,33 @@
...
-            .await?
-            .bytes()
-            .await?;
+            .send().await?
+            .bytes().await?;
         let channel = Channel::read_from(&resp[..])?;
         for item in channel.items {
-            let date_str = item.title.unwrap();
-            let date_str = &date_str[date_str.find(' ').unwrap()+1..];
+            let date_str = &item.title.unwrap()[date_str.find(' ').unwrap()+1..];
```
Comment (lajp) [inline src/util/parse_lykeion.rs:18]: I don't think this is how it works...

**Review by lajp — CHANGES_REQUESTED** [review]
Won't compile, try again

File: src/util/parse_lykeion.rs (hunk 5 above)
Comment (romeq) [reply src/util/parse_lykeion.rs:18]: yesyes that was actually only one I was unsure about

Comment (romeq) [reply src/util/parse_lykeion.rs:18]: Why tf won't it work though

Comment (lajp) [reply src/util/parse_lykeion.rs:18]: you are referencing date_str before it is assigned (as `date_str.find(...`)

Comment (romeq) [reply src/util/parse_lykeion.rs]: yes I figured it out

Comment (romeq) [reply src/util/parse_lykeion.rs]: I didn't see that other `date_str` there

**Review by lajp — APPROVED** [review]
Now works

**Comment by lajp** [comment]
LGTM, merging now, hopefully shit stays clear from the fan

**Comment by romeq** [comment]
Hope so.

**Pull request merged**